### PR TITLE
Add vietnamese language

### PR DESCRIPTION
**Pull request by phungdoanh**

### CODE REVIEW

File: lib/lang/vi.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  while(i !== m.length &&
+        i !== b.length &&
+        m.charCodeAt(i) === b.charCodeAt(i))
+    ++i;
+
+  while(i && m.charCodeAt(i - 1) !== 32)
+    --i;
+
+  return a + joiner + b.slice(i);
+}
```
Comment (ironwallaby) [inline lib/lang/vi.js]: Does Vietnamese actually make use of this, or can it be removed?

File: lib/lang/vi.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+function strip_prefix(period) {
+  return period.slice(0, 9) === "suốt đêm" ? period.slice(4) :
+    period.slice(0, 7) === "trong " ? period.slice(7) :
+      period;
+}
```
Comment (ironwallaby) [inline lib/lang/vi.js]: Does Vietnamese actually make use of this, or can it be removed?

File: lib/lang/vi.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    );
+  },
+  "through": function(a, b) {
+    return join_with_shared_prefix(a, b, " cho tới ");
+  },
+  "with": "$1, với $2",
+  "range": "$1\u2013$2",
+  "parenthetical": function(a, b) {
+    return a + " (" + b + (a === "mixed precipitation" ? " of snow)" : ")");
+  },
```
Comment (ironwallaby) [inline lib/lang/vi.js]: `"of snow"` is not translated here! Either it should be, or else this should simply be `"$1 ($2)"`.

File: lib/lang/vi.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  // "and". (This is a very crude bastardization of proper English titling
+  // rules, but it is adequate for the purposes of this module.)
+  "title": function(str) {
+    return str.replace(
+      /\b(?:a(?!nd\b)|c(?!m\.)|i(?!n\.)|[^\Waci])/g,
+      function(letter) {
+        return letter;
+      }
+    );
+  },
```
Comment (ironwallaby) [inline lib/lang/vi.js]: This should simply be `"$1"`.

File: lib/lang/vi.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  "sentence": function(str) {
+    // Capitalize.
+    str = str.charAt(0).toUpperCase() + str.slice(1);
+
+    // Add a period if there isn't already one.
+    if(str.charAt(str.length - 1) !== ".")
+      str += ".";
+
+    return str;
+  },
```
Comment (ironwallaby) [inline lib/lang/vi.js]: Is this the correct capitalization rule for Vietnamese?

File: lib/lang/vi.js (hunk 1 above)
Comment (phungdoanh) [reply lib/lang/vi.js]: I will remove it in new version

File: lib/lang/vi.js (hunk 2 above)
Comment (phungdoanh) [reply lib/lang/vi.js]: I will remove it in new version

File: lib/lang/vi.js (hunk 3 above)
Comment (phungdoanh) [reply lib/lang/vi.js]: I will fix it

File: lib/lang/vi.js (hunk 4 above)
Comment (phungdoanh) [reply lib/lang/vi.js]: I will fix it

File: lib/lang/vi.js (hunk 5 above)
Comment (phungdoanh) [reply lib/lang/vi.js]: Yes, this is  the correct capitalization rule for Vietnamese

**Comment by ironwallaby** [comment]
Looks great. There's a few minor cleanup things to do, but I can manage them myself. Thanks!

I'll integrate this here, and we'll release it as soon as we do our next production deploy (likely next week).

**Pull request merged**